### DISCUSSION TITLE
Migrating from django-tenant-schemas

### DISCUSSION THREAD

**Discussion by tomturner** [django-tenants/django-tenants#537] [General]
Anyone wanting to talk about migrating from django-tenant-schemas can be talked about here

**Comment by tomturner** [comment]
The main differnces with django-tenant-schemas and django-tenant are.

- The Domain table
- The middleware

The process I would do is

1. Swap all references from django tenant schemas to django tenants in the setting.py 
2. add` domain_url = models.CharField(max_length=128, unique=True) ` to your tenant modal
3. add the new domain model
4. make migrations
5. add data migration to copy domain_url to new domain table
6. remove domain_url from tenant model
7. make migrations
8. migrate
9. Test

Good luck

**Reply by asixela** [reply]
Great. Thanks

**Reply by awaazai** [reply]
@tomturner in step two I think you mean tenant _model_, correct?

> 2. add`domain_url = models.CharField(max_length=128, unique=True)` to your tenant **model**

**Comment by delizin** [comment]
I'm rewriting an old project that used django-tenant-schemas. I used to use the Django admin for creating tenants, but I don't understand how creating Tenants with Django admin is intended to work with the new Domain table. It appears that creating a new Tenant this way doesn't automatically create a corresponding Domain. 

My solution would be to add a domain_url field to my Tenant model and then use a post_schema_sync signal to create a Domain with that domain_url. Is this the proper method or am I going about this the wrong way?

Thank you
